src/main/java/net/openhft/chronicle/threads/Threads.java

```diff
@@ -163,12 +163,9 @@ public static void interrupt(ExecutorService service) {
 
     static void forEachThread(ExecutorService service, Consumer<Thread> consumer) {
         try {
-            final Set workers;
-            if (!(service instanceof ThreadPoolExecutor)) {
-                workers = Jvm.getValue(resolveDelegatedExecutorServices(service), "workers");
-            } else {
-                workers = Jvm.getValue(service, "workers");
-            }
+            if (!(service instanceof ThreadPoolExecutor))
+                service = resolveDelegatedExecutorServices(service);
+            final Set workers = Jvm.getValue(service, "workers");
```

Comment:
I think we need to abort at this point if `resolveDelegatedExecutorService()` doesn't yield a `ThreadPoolExecutor`.

Reply:
Sorry, one line up, before we attempt to get "workers"

Reply:
I'm pretty sure that's a bug I introduced when I made that change intitially

Reply:
Because "mainLock" and "workers" are both private fields of ThreadPoolExecutor

Reply:
yes - done